### PR TITLE
JS debugging by using DevTools

### PR DESCRIPTION
**Pull request by eduard13**
## Purpose of this PR

This PR describes another way of JS debugging by using **DevTools**, by providing the needed steps and using a small uiComponent as sample.

## Affected URLs
- https://devdocs.magento.com/guides/v2.1/ui_comp_guide/troubleshoot/ui_comp_troubleshoot_js.html
- https://devdocs.magento.com/guides/v2.2/ui_comp_guide/troubleshoot/ui_comp_troubleshoot_js.html
- https://devdocs.magento.com/guides/v2.3/ui_comp_guide/troubleshoot/ui_comp_troubleshoot_js.html

whatsnew
Added instructions for [debugging a UI component](https://devdocs.magento.com/guides/v2.3/ui_comp_guide/troubleshoot/ui_comp_troubleshoot_js.html) using Chrome DevTools.

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.1/ui_comp_guide/troubleshoot/ui_comp_troubleshoot_js.md [hunk 1]
```diff
@@ -64,6 +64,47 @@ var fieldName = ko.contextFor($0).$data;
...
+
+In the previous image, we can see three zones:
+
+1. The `Resources` zone lists all the files as HTML, JavaScript, CSS.
+2. The `Source` zone shows the source code of any selected file.
+3. The `Information and control` zone is for debugging.
+
+### Breakpoints
+
+A **breakpoint** is a point of code where the debugger will automatically pause the JavaScript execution process.
```
Comment (rogyar) [inline guides/v2.1/ui_comp_guide/troubleshoot/ui_comp_troubleshoot_js.md]: I would say it's a line of code instead of point of code

Comment (eduard13) [reply guides/v2.1/ui_comp_guide/troubleshoot/ui_comp_troubleshoot_js.md]: @rogyar the suggestion has been applied.
Thank you.

**Comment by rogyar** [comment]
From the technical standpoint, the provided update is correct

**Comment by rogyar** [comment]
@eduard13 thank you for the cool tutorial

File: guides/v2.2/ui_comp_guide/troubleshoot/ui_comp_troubleshoot_js.md [hunk 2]
```diff
@@ -61,6 +61,47 @@ var fieldName = ko.contextFor($0).$data;
...
+3. The `Information and control` zone is for debugging.
+
+### Breakpoints
+
+A **breakpoint** is a line of code where the debugger will automatically pause the JavaScript execution process.
+For setting a breakpoint, we should click right on the code line number, like in the next image.
+
+![Breakpoints]({{ site.baseurl }}/common/images/debugging-breakpoints.png)
+
+We can always find a list of breakpoints in the right pane. That’s useful when we have many breakpoints in various files.
```
Comment (jeff-matthews) [inline guides/v2.2/ui_comp_guide/troubleshoot/ui_comp_troubleshoot_js.md]:
```suggestion
You can always find a list of breakpoints in the right panel, which is useful when you have many breakpoints in various files.
```

File: guides/v2.2/ui_comp_guide/troubleshoot/ui_comp_troubleshoot_js.md [hunk 3]
```diff
@@ -61,6 +61,47 @@ var fieldName = ko.contextFor($0).$data;
 console.log(fieldName.name); // product_form.product_form.product-details.container_name.name
 ```
 
+## Debugging using Chrome DevTools
+
+All modern browsers support “debugging” – a special UI in developer tools that makes finding and fixing errors much easier. 
+
+### Sources panel UI
+
+[DevTools] provides a lot of different tools for different tasks, and the **Sources** panel is where we debug JavaScript.
```
Comment (jeff-matthews) [inline guides/v2.2/ui_comp_guide/troubleshoot/ui_comp_troubleshoot_js.md]:
```suggestion
[DevTools] provides a lot of different tools for different tasks, but the **Sources** panel is where you debug JavaScript.
```

File: guides/v2.2/ui_comp_guide/troubleshoot/ui_comp_troubleshoot_js.md [hunk 4]
```diff
@@ -61,6 +61,47 @@ var fieldName = ko.contextFor($0).$data;
...
+
+1. Open the required page in Chrome.
+2. Turn on developer tools with F12 (Windows, Linux) or Cmd+Opt+I (Mac).
+3. Select the `Sources` pane.
```
Comment (jeff-matthews) [inline guides/v2.2/ui_comp_guide/troubleshoot/ui_comp_troubleshoot_js.md]:
```suggestion
3. Click the `Sources` tab.
```

File: guides/v2.2/ui_comp_guide/troubleshoot/ui_comp_troubleshoot_js.md [hunk 5]
```diff
@@ -61,6 +61,47 @@ var fieldName = ko.contextFor($0).$data;
...
+In the previous image, we can see three zones:
+
+1. The `Resources` zone lists all the files as HTML, JavaScript, CSS.
+2. The `Source` zone shows the source code of any selected file.
+3. The `Information and control` zone is for debugging.
+
+### Breakpoints
+
+A **breakpoint** is a line of code where the debugger will automatically pause the JavaScript execution process.
+For setting a breakpoint, we should click right on the code line number, like in the next image.
```
Comment (jeff-matthews) [inline guides/v2.2/ui_comp_guide/troubleshoot/ui_comp_troubleshoot_js.md]:
```suggestion
To set a breakpoint, right-click on the code line number (as shown in the next image).
```

**Comment by eduard13** [comment]
Hi @jeff-matthews thank you for your suggestions, they were applied for both versions.

**Comment by eduard13** [comment]
Hi @jeff-matthews is there anything else that I should do here?
Thank you.

**Comment by jeff-matthews** [comment]
@eduard13, nope! I'll test and merge.

**Comment by jeff-matthews** [comment]
running tests

**Pull request merged**